tests/test_crf.py

```diff
@@ -476,3 +476,186 @@ def assert_close(actual, expected):
 
     def assert_close(actual, expected):
         torch.testing.assert_allclose(actual, expected, atol=1e-12, rtol=1e-6)
+
+
```

Comment:
Please add `# TODO reduce duplicated code below, maybe with pytest.mark.parametrize` here (this todo is for me, you don't need to do anything about it).

Reply:
Please check that I added the comment in the intended place.